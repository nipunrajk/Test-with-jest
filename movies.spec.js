import movies from "./movies";

describe("favourite movies", () => {
  let myMovies = [];
  beforeEach(() => {
    myMovies = [
      {
        title: "chapter 3",
        rate: 9,
      },
    ];
  });
  test("can add a movie", () => {
    movies.add(myMovies, "john wickr");
    expect(myMovies).toMatchSnapshot();
  });

  test("rate a movie", () => {
    movies.rate(myMovies[0], 5);
    expect(myMovies).toMatchSnapshot();
  });
});
