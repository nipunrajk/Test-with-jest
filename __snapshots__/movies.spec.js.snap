// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`favourite movies can add a movie 1`] = `
[
  {
    "rate": 9,
    "title": "chapter 3",
  },
  {
    "rate": null,
    "title": "john wick",
  },
]
`;

exports[`favourite movies rate a movie 1`] = `
[
  {
    "rate": 5,
    "title": "chapter 3",
  },
]
`;
